### PR TITLE
Changes rtl check in ChatMessages class

### PR DESCRIPTION
**Pull request by MartinSchoeler**
instead of checking if the text is rtl in every input, it checks in the first input only 
@RocketChat/core

### CODE REVIEW

File: packages/rocketchat-authorization/server/startup.coffee [hunk 1]
```diff
@@ -41,9 +41,9 @@ Meteor.startup ->
 		{ _id: 'set-owner', roles : ['admin', 'owner'] }
 		{ _id: 'unarchive-room', roles : ['admin'] }
 		{ _id: 'view-c-room', roles : ['admin', 'user', 'bot'] }
-		{ _id: 'view-d-room', roles : ['admin', 'user'] }
+		{ _id: 'view-d-room', roles : ['admin', 'user', 'bot'] }
 		{ _id: 'view-full-other-user-info', roles : ['admin'] }
-		{ _id: 'view-history', roles : ['user'] }
+		{ _id: 'view-history', roles : ['admin', 'user', 'bot'] }
```
Comment (geekgonecrazy) [inline packages/rocketchat-authorization/server/startup.coffee]: Looks like some un-related changes made it in with this?

File: packages/rocketchat-ui/lib/chatMessages.coffee [hunk 2]
```diff
@@ -378,8 +378,9 @@ class @ChatMessages
 		else if k is 75 and ((navigator?.platform?.indexOf('Mac') isnt -1 and event.metaKey and event.shiftKey) or (navigator?.platform?.indexOf('Mac') is -1 and event.ctrlKey and event.shiftKey))
 			RoomHistoryManager.clear rid
 
-	valueChanged: (rid, event) ->
-		this.determineInputDirection()
+	valueChanged: (rid, event, input) ->
+		if input.lenght is 1
```
Comment (geekgonecrazy) [inline packages/rocketchat-ui/lib/chatMessages.coffee]: Not sure how this didn't throw errors on you locally.  :)  But need to change this to: `input.length is 1` to be valid

**Comment by geekgonecrazy** [comment]
Left a few notes inline.   Is this addressing a particular issue?  It looks like it might address an issues i've been seeing.  Just want to confirm which you're addressing.  :smile:

**Comment by rodrigok** [comment]
Hi @MartinSchoeler, I think you are reverting the improvement did on [this PR](https://github.com/RocketChat/Rocket.Chat/pull/3730?ts=2)

File: packages/rocketchat-ui/lib/chatMessages.coffee (hunk 2 above)
Comment (MartinSchoeler) [inline packages/rocketchat-ui/lib/chatMessages.coffee]: Ohhh how didn't i see that! XD

File: packages/rocketchat-ui/lib/chatMessages.coffee [hunk 3]
```diff
@@ -378,8 +378,9 @@ class @ChatMessages
 		else if k is 75 and ((navigator?.platform?.indexOf('Mac') isnt -1 and event.metaKey and event.shiftKey) or (navigator?.platform?.indexOf('Mac') is -1 and event.ctrlKey and event.shiftKey))
 			RoomHistoryManager.clear rid
 
-	valueChanged: (rid, event) ->
-		this.determineInputDirection()
+	valueChanged: (rid, event, input) ->
+		if this.input.value.length is 1
```
Comment (marceloschmidt) [inline packages/rocketchat-ui/lib/chatMessages.coffee]: This won't work! You can't use `this.input` there.

Comment (MartinSchoeler) [inline packages/rocketchat-ui/lib/chatMessages.coffee]: I tested here and worked, why do you think it wont work? also the derermineInputDirection() method uses this.input too.

Comment (marceloschmidt) [inline packages/rocketchat-ui/lib/chatMessages.coffee]: My bad. I didn't realize `this.input` is defined inside the class.

Comment (MartinSchoeler) [inline packages/rocketchat-ui/lib/chatMessages.coffee]: :smile:

File: packages/rocketchat-authorization/server/startup.coffee (hunk 1 above)
Comment (MartinSchoeler) [inline packages/rocketchat-authorization/server/startup.coffee]: Oh i forgot about these! removing now

**Pull request merged**